### ISSUE TITLE
Filter HTML diagnostics not needed for .razor

### ISSUE THREAD

**Issue by TanayParikh** [dotnet/aspnetcore#28654]
~~Blocked on: Filtering HTML diagnostics needs diagnostic code annotations. 
Tracked via: https://dev.azure.com/devdiv/DevDiv/_workitems/edit/1257401~~

Unblocked & ready to go now.

**Comment by TanayParikh** [comment]
Nested warnings: https://github.com/dotnet/aspnetcore/issues/30358

**Comment by TanayParikh** [comment]
> - Missing head/body/html in Blazor documents. SHould be suppressed since most files will not have an <html> / <head> / <body> tag
> - Any diagnostic on aa TagHelper element should be ignored. These typically represent themselves as "missing href" etc. because the HTML editor doesn't understand that one of our special attributes "generates" an href behind the scenes.
- block duplicate diagnostics. Ex. for unclosed tag we don't want RZ* diagnostics and WTE diagnostics

**Comment by TanayParikh** [comment]
> Missing head/body/html in Blazor documents. SHould be suppressed since most files will not have an / / tag
> Any diagnostic on aa TagHelper element should be ignored. These typically represent themselves as "missing href" etc. because the HTML editor doesn't understand that one of our special attributes "generates" an href behind the scenes.

Not seeing these actually materialize in the first place, if anyone has a specific example, please do let me know.

**Comment by TanayParikh** [comment]
> Any diagnostic on aa TagHelper element should be ignored. These typically represent themselves as "missing href" etc. because the HTML editor doesn't understand that one of our special attributes "generates" an href behind the scenes.

https://github.com/dotnet/aspnetcore/issues/5014

This has been addressed via https://github.com/dotnet/aspnetcore-tooling/blob/63c2ea0fd53fb24b5ab0c0d4ad01e389c4cd9538/src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/RazorDiagnosticsEndpoint.cs#L232-L237

If that's true, we ignore diagnostics there. Specifically, in this case, we see a MarkupTagHelperAttributeValue and hence ignore the missing href diagnostic.
